addons/project/tests/test_project_flow.py

```diff
@@ -424,3 +424,14 @@ def test_project_sub_task_copy_without_archive_user(self):
         self.user_projectmanager.action_archive()
         task_1_copy = self.task_1.copy()
         self.assertFalse(task_1_copy.child_ids.user_ids)
+
+    def test_task_email_context_with_subtitles(self):
+        task = self.env['project.task'].create({
+            'name': 'Task',
+            'user_ids': [Command.set([self.user_projectuser.id])],
+            'project_id': self.project_goats.id,
+        })
+        self.assertFalse(self.project_goats.message_follower_ids)
+        self.assertEqual(self.project_goats.privacy_visibility, 'followers')
```

Comment:
@ppr-odoo 
You don't have the access error because the project is in cache because we read some fields to do the asserts with odooBot user. But if we invalidate the recordset to make sure we do not have the recordset in the cache, then you will see the sudo is actually needed.
```suggestion
        self.assertEqual(self.project_goats.privacy_visibility, 'followers')
        self.project_goats.invalidate_recordset()
```

Reply:
> @ppr-odoo You don't have the access error because the project is in cache because we read some fields to do the asserts with odooBot user. But if we invalidate the recordset to make sure we do not have the recordset in the cache, then you will see the sudo is actually needed.

Yes, but I did not face an access error in this flow.

Steps:
  - Create a project user.
  - Create a project with privacy_visibility set to followers.
  - Create a task and assign it to the project user.
  - Log in as the project user.
  - Send a message.

Reply:
To me, normally, you should have the issue if the project is not in the cache as I mentioned before. 🤔

Reply:
> To me, normally, you should have the issue if the project is not in the cache as I mentioned before. 🤔

I sent a message with an attachment on the Vidange server. Thanks.